app/models/post.rb

```diff
@@ -4,4 +4,15 @@ class Post < ApplicationRecord
 
   belongs_to :day
   belongs_to :user
+
+  def self.with_matching_group(user_group_id)
+    all_posts = joins(:day).select('posts.*, days.group_id as group')
+    all_posts.find_all {|post| post.group == user_group_id}
+  end
```

Comment:
@jklich151 @d-atkins @mronauli 

This class method first gets all the posts with the associated group_id, found through the day, added as an attribute. Then, I use those posts to filter for the ones that match the current user's group id.

Couple of questions: 
1) Asking for support refactoring the second part of that method to activerecord.

2) If users have group and posts have days which also have groups; which group id are we trying to reference in here? I'm asking because a user and day can have different groups; but both be on the same post when created.

Reply:
@eamouzou @d-atkins @jklich151 would like clarity on question 2 as well. great opportunity for group collab/discussion

Reply:
I think some things we discussed might have been lost/quietly changed while I was writing user stories, so sorry for the confusion. I would pass all of a user's posts to be displayed in their profile/posts page. So in the controller, I would just pass `@posts = current_user.posts`, no AR. That way, if a user graduates to the next mod, they can still see the posts they've made before (maybe that contradicts what we talked about? I don't remember, but I think it makes sense to show them all)

A post should be referencing the group that belongs to its `day`, because that reflects the mod in which that post was made.